rules/renaming/tests/Rector/FileWithoutNamespace/PseudoNamespaceToNamespaceRector/PseudoNamespaceToNamespaceRectorTest.php

```diff
@@ -42,4 +48,65 @@ protected function getRectorsWithConfiguration(): array
             ],
         ];
     }
+
+    /** @noinspection PhpMissingReturnTypeInspection */
+    // @phpstan-ignore-next-line
+    private function getAutoloadClassNameAssertion()
+    {
+        return new class ($this) { // @phpstan-ignore-line
+            // ref: https://www.php.net/manual/en/language.oop5.basic.php
+            private const CLASS_NAME = '/^
+                # valid PHP class-name (incl. optional leading \\ name-space separator)
+                \\\\?
+                [a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*
+                (?: \\\\ [a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]* )*
+                $/Dx';
+
+            /**
+             * @var TestCase
+             */
+            private $testCase;
+
+            /**
+             * @var string[]
+             */
+            private $sequence;
+
+            /**
+             * @param TestCase $testCase
+             * @noinspection InterfacesAsConstructorDependenciesInspection
+             */
+            public function __construct(TestCase $testCase)
+            {
+                $this->testCase = $testCase;
+                \spl_autoload_register($this, true ,true);
+            }
+
+            public function __invoke(string $className): void
+            {
+                $this->sequence[] = $className;
+            }
+
+            /**
+             * @param SmartFileInfo $fileInfo
+             * @throws ExpectationFailedException
+             */
+            public function assert(SmartFileInfo $fileInfo): void
+            {
+                foreach ($this->sequence as $className) {
+                    $this->testCase::assertMatchesRegularExpression(
+                        self::CLASS_NAME,
+                        $className,
+                        $fileInfo->getBasename() . ' class-name validity'
+                    );
+                }
+                $this->sequence = [];
+            }
+
+            public function __destruct()
+            {
+                \spl_autoload_unregister($this);
```

Comment:
This must be handled without hacking autoload, so it's easy to maintain in years to come

Reply:
Yes sure this must be stable as the foundations of mountains. Therefore it would help me if you could elaborate a bit where this spl_autoload_register / spl_autoload_unregister scheme poses a maintenance burden? Actually I can not imagine that it will go away nor that composer drops it. @TomasVotruba

Reply:
Autoload is handled by composer, so we can autoload it by putting in psr-4 scope or not autoloadit by putting anywhere else. No need to create 2nd composer :)

Reply:
Yes, sure. But this is all not happening with the change-set :) It just keeps a list of class-names to assert it later ...

Reply:
See https://github.com/rectorphp/rector/pull/5040/files